test/common_test/construct_empty_world.cc

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <gtest/gtest.h>
+
+#include <gz/common/Console.hh>
+
+#include "../helpers/TestLibLoader.hh"
+
+#include <gz/physics/ConstructEmpty.hh>
+#include <gz/physics/FindFeatures.hh>
+#include <gz/physics/GetEntities.hh>
+#include <gz/physics/RequestEngine.hh>
+#include <gz/physics/RemoveEntities.hh>
+
+#include "gz/physics/BoxShape.hh"
+
+// The features that an engine must have to be loaded by this loader.
+using Features = gz::physics::FeatureList<
+  gz::physics::GetEngineInfo,
+  gz::physics::GetWorldFromEngine,
+  gz::physics::ConstructEmptyWorldFeature,
+  gz::physics::ConstructEmptyModelFeature,
+  gz::physics::GetModelFromWorld,
+  gz::physics::GetLinkFromModel,
+  gz::physics::ConstructEmptyLinkFeature,
+  gz::physics::AttachBoxShapeFeature,
+  gz::physics::GetShapeFromLink,
+  gz::physics::GetBoxShapeProperties,
+  gz::physics::RemoveEntities,
+  gz::physics::GetNestedModelFromModel,
+  gz::physics::ConstructEmptyNestedModelFeature
```

Comment:
future TODO: split up the tests and features into multiple, granular tests and `FeatureList`s to allow be more useful as people are incrementally implementing a plugin. 

This doesn't need to be done now

Reply:
I openned this draft PR https://github.com/gazebosim/gz-physics/pull/364 so we can discuss the ideas there and unblock this one